DNN Platform/Library/Entities/Modules/Settings/SettingsRepository.cs

```diff
@@ -164,7 +164,13 @@ private void SaveSettings(int portalId, ModuleInfo moduleContext, T settings)
                     }
                 }
             });
-            DataCache.SetCache(this.CacheKey(portalId, moduleContext?.TabModuleID ?? -1), settings);
+
+            DataCache.ClearCache(this.CacheKeyPortalPrefix(portalId));
+            DataCache.SetCache(this.CacheKey(portalId, -1), settings);

```

Comment:
Maybe I'm missing something, but shouldn't this only be happening if the `moduleContext` is null?

If the module context is null, then we're talking about global portal settings, otherwise module-specific settings. Now the module-specific settings can be set in the portal settings.

Reply:
A settings class can contain both module settings and portal settings, so when a settings class is updated, it needs to invalidate all of the other settings classes for that portal.

This is caching module-specific settings with the non-module-specific cache key, under the assumption that any code which is only passing in a portal ID will know to not use the values of the module-specific properties. Perhaps this is not a safe-enough (or efficient-enough) optimization and should be removed.

Reply:
I've pushed a commit to remove that optimization